plugins/experimental/ssl_session_reuse/src/session_process.cc

```diff
@@ -0,0 +1,252 @@
+/** @file
+
+  session_process.cc - encrypt and decrypt sessions
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+ */
+
+#include <math.h>
+#include <string.h>
+#include <errno.h>
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+#include <openssl/rand.h>
+#include <ts/ts.h>
+
+#include "session_process.h"
+#include "ssl_utils.h"
+
+#include "common.h"
+
+const unsigned char salt[] = {115, 97, 108, 117, 0, 85, 137, 229};
+
+int
+encrypt_session(const char *session_data, int32_t session_data_len, const unsigned char *key, int key_length,
+                std::string &encrypted_data)
+{
+  size_t len_all = 0;
+  size_t offset  = 0;
+  char *pBuf     = nullptr;
+
+  int encrypted_buffer_size    = 0;
+  int encrypted_msg_len        = 0;
+  unsigned char *encrypted_msg = NULL;
+  int ret                      = 0;
+
+  offset  = 0;
+  len_all = sizeof(int64_t) + sizeof(int32_t) + session_data_len;
+
+  pBuf = new char[len_all];
```

Comment:
Why not:

std::unique_ptr<char []> pBuf(new char[len_all]);

Reply:
Or `std::string pBuff; pBuff.resize(len_all);`.